modules/op-pipelines-concepts.adoc

```diff
@@ -0,0 +1,24 @@
+// Module included in the following assemblies:
+//
+// * openshift_pipelines/understanding-openshift-pipelines.adoc
+
+[id='op-pipelines-concepts_{context}']
+= {op-title} concepts
+
+{op-title} provide the following set of standard CRDs that act as the building blocks from which you can assemble a CI/CD pipeline for your application:
+
+Task:: A Task is the smallest configurable unit in a Pipeline. It is essentially a function of inputs and outputs that form the Pipeline build. It can run individually or as a part of a Pipeline. A Pipeline includes one or more Tasks, where each Task consists of one or more steps. Steps are a series of commands that are sequentially executed by the Task.
+
+Pipeline:: A Pipeline consists of a series of Tasks that are executed to construct complex workflows that automate the build, deployment, and delivery of applications. It is a collection of PipelineResources, Parameters, and one or more Tasks. A Pipeline interacts with the outside world by using PipelineResources, which are added to Tasks as inputs and outputs.
+
+PipelineRun:: A PipelineRun is the running instance of a Pipeline. PipelineRun initiates a Pipeline and manages the creation of a TaskRun for each Task being executed in the Pipeline.
+
+TaskRun:: A TaskRun is auto-created by a PipelineRun for each Task in a Pipeline. It is a result of running an instance of a Task in a Pipeline. It can also be manually created if a Task runs outside of a Pipeline.
+
+PipelineResource:: A PipelineResource is an object that is used as an input and output for pipeline Tasks. For example, if an input is a Git repository and an output is a container image built from that Git repository, these are both classified as PipelineResources. PipelineResources currently support Git resources, Image resources, Cluster resources, Storage Resources and CloudEvent resources.
+
+Trigger:: A Trigger captures an external event, such as Git pull request, and processes the event payload received to extract key pieces of information. This extracted information is then mapped to a set of predefined parameters, which trigger a series of tasks that may involve creation and deployment of Kubernetes resources. Using Triggers along with Pipelines, you can create a full-fledged CI/CD systems where the execution is defined entirely through Kubernetes resources.
+
+Condition:: A Condition refers to a validation or check which is executed before a Task is run in your Pipeline. Conditions are like `if` statements which perform logical tests, with a return value of `True` or `False`. A Task is executed if all Conditions return `True`, else if any of the Conditions fails, the Task and all subsequent Tasks are skipped. By using Conditions in your Pipeline, you can create complex workflows covering multiple scenarios.
```

Comment:
check, which
fails/fail
all the subsequent
You can use Conditions in your Pipeline to create...?